Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -2401,7 +2401,9 @@ public static InputEventListener onEvent
         /// <seealso cref="ButtonControl.isPressed"/>
         /// <seealso cref="onEvent"/>
         public static IObservable<InputControl> onAnyButtonPress =>
-            onEvent.Select(e => e.GetFirstButtonPressOrNull()).Where(c => c != null);
+            onEvent
+                .Where(evt => evt.type == StateEvent.Type || evt.type == DeltaStateEvent.Type)
```

Comment:
Minor but this will perform a null check twice. Better to fetch the type here only once.